### PR TITLE
Added new API functionalities

### PR DESCRIPTION
**Pull request by Cowplant-Simmer-Collin**
New API calls for
- Avatar info
- Lot info
- Neighborhood info
- Bulletin info
- Election info

### CODE REVIEW

File: TSOClient/FSO.Server.Api.Core/Controllers/LotInfoController.cs [hunk 1]
```diff
@@ -144,6 +144,262 @@ public IActionResult GetJSON(int shardid, uint id)
...
+        //get online lots
+        [HttpGet]
+        [Route("userapi/city/{shardid}/lots/online.json")]
+        public IActionResult GetOnline(int shardid)
+        {
+            var api = Api.INSTANCE;
+
+            using (var da = api.DAFactory.Get())
+            {
+                var Lots = da.LotClaims.AllLocations(shardid);
```
Comment (riperiperi) [inline TSOClient/FSO.Server.Api.Core/Controllers/LotInfoController.cs]: Local variables have `lowerCamelCase`.

File: TSOClient/FSO.Server.Api.Core/Controllers/ElectionInfoController.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                var ElectionCycle = da.Elections.GetCycle((uint)Nhood.election_cycle_id);
+                
+                var ElectionCandidates = new List<DbElectionCandidate>();
+                if (ElectionCycle.current_state == Database.DA.Elections.DbElectionCycleState.election)
+                {
+                    ElectionCandidates = da.Elections.GetCandidates(ElectionCycle.cycle_id, Database.DA.Elections.DbCandidateState.running);
+                }
+                if (ElectionCycle.current_state == Database.DA.Elections.DbElectionCycleState.ended)
+                {
+                    ElectionCandidates = da.Elections.GetCandidates(ElectionCycle.cycle_id, Database.DA.Elections.DbCandidateState.won);
```
[the remark this answers is not in the document]
Comment (Cowplant-Simmer-Collin) [reply TSOClient/FSO.Server.Api.Core/Controllers/ElectionInfoController.cs]: Yes, that has been tested 😄

File: TSOClient/FSO.Server.Api.Core/Controllers/LotInfoController.cs [hunk 3]
```diff
@@ -144,6 +144,262 @@ public IActionResult GetJSON(int shardid, uint id)
...
+                    category = Lot.category,
+                    created_date = Lot.created_date,
+                    description = Lot.description,
+                    location = Lot.location,
+                    name = Lot.name,
+                    neighborhood_id = Lot.neighborhood_id,
+                    owner_id = Lot.owner_id,
+                    shard_id = Lot.shard_id,
+                    skill_mode = Lot.skill_mode,
+                    roommates = roomies
```
[the remark this answers is not in the document]
Comment (Cowplant-Simmer-Collin) [reply TSOClient/FSO.Server.Api.Core/Controllers/LotInfoController.cs]: It only selects the avatar id from the roommates in the database call, so if sensitive information will be added it won't be added to the JSON table. 👍

**Comment by Cowplant-Simmer-Collin** [comment]
Thanks for all the feedback, I really appreciate it. I worked on it and committed it again with all the fixes and extra optimizations

File: TSOClient/FSO.Server.Api.Core/Controllers/NhoodInfoController.cs [hunk 4]
```diff
@@ -17,133 +17,117 @@ namespace FSO.Server.Api.Core.Controllers
...
-                return ApiResponse.Json(HttpStatusCode.OK, NhoodsJSON);
+                var nhoodsJson = new JSONNhoods();
+                nhoodsJson.neighborhoods = nhoodJson;
+                return ApiResponse.Json(HttpStatusCode.OK, nhoodsJson);
             }
         }
         [HttpGet]
-        [Route("userapi/neighborhoods/id/{Nhoodid}.json")]
-        public IActionResult GetByID(uint Nhoodid)
+        [Route("userapi/neighborhoods/{NhoodId}")]
```
Comment (riperiperi) [inline TSOClient/FSO.Server.Api.Core/Controllers/NhoodInfoController.cs]: Case mismatch with parameter?

Comment (Cowplant-Simmer-Collin) [reply TSOClient/FSO.Server.Api.Core/Controllers/NhoodInfoController.cs]: Must have overlooked that typo, thanks for pointing it out 👍

**Comment by riperiperi** [comment]
LGTM, very nice work!

**Pull request merged**